recommend/imageHandler.go

```diff
@@ -0,0 +1,362 @@
+	f, err := os.Open(filepath.Clean(fname))
+	if err != nil {
+		log.WithFields(log.Fields{
+			"file": fname,
+		}).Fatal("open file failed")
+	}
+	defer closeCheckErr(f, fname)
+	return io.ReadAll(f)
+}
+
+func readManifest(img *ImageInfo, manifest string) {
+	// read manifest file
+	barr, err := getFileBytes(manifest)
+	if err != nil {
+		log.WithError(err).Fatal("manifest read failed")
+	}
+	var manres []map[string]interface{}
+	err = json.Unmarshal(barr, &manres)
+	if err != nil {
+		log.WithError(err).Fatal("manifest json unmarshal failed")
+	}
+	if len(manres) < 1 {
+		log.WithFields(log.Fields{
+			"len":     len(manres),
+			"results": manres,
+		}).Fatal("expecting atleast one config in manifest!")
+	}
+
+	var man map[string]interface{}
+	for _, man = range manres {
+		if man["RepoTags"] != nil {
+			break
+		}
+	}
+
+	// read config file
+	config := filepath.Join(tempDir, man["Config"].(string))
+	barr, err = getFileBytes(config)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"config": config,
+		}).Fatal("config read failed")
+	}
+	var cfgres map[string]interface{}
+	err = json.Unmarshal(barr, &cfgres)
+	if err != nil {
+		log.WithError(err).Fatal("config json unmarshal failed")
+	}
+	img.Arch = cfgres["architecture"].(string)
+	img.OS = cfgres["os"].(string)
+	for _, tag := range man["RepoTags"].([]interface{}) {
+		img.RepoTags = append(img.RepoTags, tag.(string))
+	}
+}
+
+type distroRule struct {
+	Distro string `json:"distro"`
+	Match  []struct {
+		Path string `json:"path"`
+	} `json:"match"`
+}
+
+//go:embed json/distro.json
+var distroJSON []byte
+
+var distroRules []distroRule
+
+func init() {
+	err := json.Unmarshal(distroJSON, &distroRules)
+	if err != nil {
+		color.Red("failed to unmarshal distro json rules")
+		log.WithError(err).Fatal("failed to unmarshal distro json rules")
+	}
+}
+
+func getDistro(img *ImageInfo) {
+	for _, d := range distroRules {
+		match := true
+		for _, m := range d.Match {
+			matches := checkForSpec(filepath.Clean(tempDir+m.Path), img.FileList)
+			if len(matches) == 0 {
+				match = false
+				break
+			}
+		}
+		if len(d.Match) > 0 && match {
+			color.Green("Distribution %s", d.Distro)
+			img.Distro = d.Distro
+			return
+		}
+	}
+}
+
+func getImageInfo(img *ImageInfo) {
+	matches := checkForSpec(filepath.Join(tempDir, "manifest.json"), img.FileList)
+	if len(matches) != 1 {
+		log.WithFields(log.Fields{
+			"len":     len(matches),
+			"matches": matches,
+		}).Fatal("expecting one manifest.json!")
+	}
+	readManifest(img, matches[0])
+
+	getDistro(img)
+}
+
+func getImageDetails(imageName string) error {
+	var img ImageInfo
+	img.Name = imageName
+	// step 1: save the image to a tar file
+	tarname := saveImageToTar(imageName)
+
+	// step 2: retrieve information from tar
+	img.FileList, img.DirList = extractTar(tarname)
+
+	// step 3: getImageInfo
+	getImageInfo(&img)
+
+	// step 4: get policy from image info
+	getPolicyFromImageInfo(&img)
```

Comment:
Essentially we are reading the tar file and appending individual file/dir to arrays.
And then doing a regex match on each element in the array using based on PreCondition?

Is that right?

Reply:
Yep, that's right